core/deployment/src/main/java/io/quarkus/deployment/index/ClassPathArtifactResolver.java

```diff
@@ -67,15 +69,15 @@ public ResolvedArtifact getArtifact(String groupId, String artifactId, String cl
         for (StoredUrl url : pathList) {
             Matcher matcher = filePatten.matcher(url.fileName);
             if (matcher.matches()) {
-                String[] groupParts = groupId.split("\\.");
-                if (url.path.getNameCount() < groupParts.length + 2) {
+                List<String> groupParts = Splitter.on('.').splitToList(groupId);
```

Comment:
It doesn't seem to be worth bringing guava into core just for this.

Reply:
Agreed, it can become a nightmare because of dependency conflicts.